### PR TITLE
chore: add form modifiers

### PR DESCRIPTION
**Pull request by shpran**
https://oat-sa.atlassian.net/browse/ADF-1779
https://oat-sa.atlassian.net/browse/ADF-1780
https://oat-sa.atlassian.net/browse/ADF-1781
https://oat-sa.atlassian.net/browse/ADF-1783

## Changes
* Added modifier for translation items to fill the unique ID element

## TODO
- [ ] Update composer dependencies

# PRs
* https://github.com/oat-sa/tao-core/pull/4094
* https://github.com/oat-sa/extension-tao-item/pull/660
* https://github.com/oat-sa/extension-tao-test/pull/476
* https://github.com/oat-sa/extension-tao-testqti/pull/2503

[ADF-1780]: https://oat-sa.atlassian.net/browse/ADF-1780?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -66,9 +66,9 @@
         "oat-sa/oatbox-extension-installer": "~1.1||dev-master",
         "naneau/semver": "~0.0.7",
         "oat-sa/generis": ">=15.36.4",
-        "oat-sa/tao-core": ">=54.22.0",
-        "oat-sa/extension-tao-item": ">=12.0.0",
-        "oat-sa/extension-tao-test": ">=16.0.0"
+        "oat-sa/tao-core": "dev-feat/ADF-1781/translations-feature as 99.99",
+        "oat-sa/extension-tao-item": "dev-feat/ADF-1779/add-translator-role as 99.99",
+        "oat-sa/extension-tao-test": "dev-feat/ADF-1779/add-translator-role as 99.99"
```
Comment (shpran) [inline composer.json:71]: - [ ] Update composer dependencies

**Review by gabrielfs7 — APPROVED** [review]
Thanks @shpran 


- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: model/Translation/Listener/ItemUpdatedEventListener.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+use oat\generis\model\data\Ontology;
+use oat\oatbox\event\Event;
+use oat\tao\model\featureFlag\FeatureFlagCheckerInterface;
+use oat\tao\model\TaoOntology;
+use oat\taoItems\model\event\ItemUpdatedEvent;
+use oat\taoQtiItem\model\Translation\Service\QtiIdentifierRetriever;
+use Psr\Log\LoggerInterface;
+
+class ItemUpdatedEventListener
```
Comment (gabrielfs7) [inline model/Translation/Listener/ItemUpdatedEventListener.php:33]: - [ ] @shpran Can you add missing tests, please?

File: model/Translation/Listener/ItemUpdatedEventListener.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        QtiIdentifierRetriever $qtiIdentifierRetriever,
+        LoggerInterface $logger
+    ) {
+        $this->featureFlagChecker = $featureFlagChecker;
+        $this->ontology = $ontology;
+        $this->qtiIdentifierRetriever = $qtiIdentifierRetriever;
+        $this->logger = $logger;
+    }
+
+    public function populateTranslationProperties(Event $event): void
```
Comment (gabrielfs7) [inline model/Translation/Listener/ItemUpdatedEventListener.php:52]: I do not think this class will handle other type of event, will it?


```suggestion
    public function populateTranslationProperties(ItemUpdatedEventListener $event): void
```

File: model/Translation/Service/QtiIdentifierRetriever.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    public function retrieve(core_kernel_classes_Resource $item): ?string
+    {
+        try {
+            $itemData = $this->qtiItemService->getDataItemByRdfItem($item);
+        } catch (Throwable $exception) {
+            $this->logger->error('An error occurred while retrieving item data: ' . $exception->getMessage());
+
+            return null;
```
Comment (gabrielfs7) [inline model/Translation/Service/QtiIdentifierRetriever.php:48]: I am not sure we should ignore this error. 🤔 

```suggestion
            throw $e;
```

File: model/Translation/Listener/ItemUpdatedEventListener.php (hunk 2 above)
Comment (gabrielfs7) [reply model/Translation/Listener/ItemUpdatedEventListener.php:32]: Closing as done

File: model/Translation/Service/QtiIdentifierRetriever.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+declare(strict_types=1);
+
+namespace oat\taoQtiItem\model\Translation\Service;
+
+use core_kernel_classes_Resource;
+use oat\taoQtiItem\model\qti\Service;
+use Psr\Log\LoggerInterface;
+use Throwable;
+
+class QtiIdentifierRetriever
```
[the remark this answers is not in the document]
Comment (shpran) [reply model/Translation/Service/QtiIdentifierRetriever.php:30]: Done ✅

File: model/Translation/Service/QtiIdentifierRetriever.php (hunk 4 above)
Comment (shpran) [reply model/Translation/Service/QtiIdentifierRetriever.php]: Done ✅

File: model/Translation/Listener/ItemUpdatedEventListener.php (hunk 3 above)
Comment (shpran) [reply model/Translation/Listener/ItemUpdatedEventListener.php]: Done ✅

**Pull request merged**